### PR TITLE
Pull changes from `main` into `release/2.8`

### PR DESCRIPTION
**Pull request by AlanRynne**
## Description

Just your average update from `main` to `release/2.8` with a twist.

Since there were merge conflicts that needed to be resolved but **not included in `main`**, I created an auxiliary branch to fix these.

### CODE REVIEW

**Review by teocomi — APPROVED** [review]

**Comment by AlanRynne** [comment]
I think there's no need to wait for Code Owner approval (@clairekuang). changes in Rhino are not critical.

If Claire has an issue with any of these changes, please bring it up and we'll deal with it accordingly!

File: Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs [hunk 1]
```diff
@@ -70,8 +81,94 @@ public ApplicationObject FamilyInstanceToNative(BuiltElements.Revit.FamilyInstan
...
+
+              if (cutVoidsParams.ElementAtOrDefault(0) != null && cutVoidsParams[0].AsInteger() == 1)
+                InstanceVoidCutUtils.AddInstanceVoidCut(Doc, wall, familyInstance);
+              if (lvlParams.ElementAtOrDefault(0) != null)
+                lvlParams[0].Set(level.Id);
+            }
+            else if (CurrentHostElement is Floor floor)
+            {
+              // TODO: support hosted elements on floors. Should be very similar to above implementation
+              Report.ConversionErrors.Add(new Exception($"Work Plane based families on floors to be supported soon"));
```
Comment (clairekuang) [inline Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: Logging should be updated to use the new applicationobject

Comment (AlanRynne) [reply Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: I created a separate issue for this so we don't forget! :D Good catch!

Comment (AlanRynne) [reply Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFamilyInstance.cs:155]: https://github.com/specklesystems/speckle-sharp/issues/1473

**Pull request merged**